cmd/guacone/cmd/analyze.go

```diff
@@ -0,0 +1,476 @@
+		uri, _ := cmd.Flags().GetBool("uri")
+		purl, _ := cmd.Flags().GetBool("purl")
+
+		metadata, _ := cmd.Flags().GetBool("metadata")
+		inclSoft, _ := cmd.Flags().GetBool("incl-soft")
+		inclDeps, _ := cmd.Flags().GetBool("incl-deps")
+		inclOccur, _ := cmd.Flags().GetBool("incl-occur")
+		namespaces, _ := cmd.Flags().GetBool("namespaces")
+		id, _ := cmd.Flags().GetBool("id")
+		test, _ := cmd.Flags().GetString("test")
+
+		var graphs []graph.Graph[string, *analyzer.Node]
+		var err error
+
+		if test == "" {
+			if err = verifyAnalyzeFlags(slsas, sboms, errSlsa, errSbom, uri, purl, id); err != nil {
+				fmt.Fprintf(os.Stderr, "Error: %s", err)
+				_ = cmd.Help()
+				os.Exit(1)
+			}
+
+			//create graphs
+			graphs, err = hasSBOMToGraph(ctx, gqlclient, sboms, AnalyzeOpts{
+				Metadata: metadata, InclSoft: inclSoft, InclDeps: inclDeps, InclOccur: inclOccur, Namespaces: namespaces, URI: uri, PURL: purl, ID: id})
+
+			if err != nil {
+				logger.Fatalf("Unable to generate graphs: %v", err)
+			}
+
+		} else {
+			graphs, err = readTwoSBOM(test)
+			if err != nil {
+				logger.Fatalf("Unable to generate graphs: %v", err)
+			}
+		}
+
+		if args[0] == "diff" {
+			analysisOne, analysisTwo, err := analyzer.HighlightAnalysis(graphs[0], graphs[1], 0)
+			if err != nil {
+				logger.Fatalf("unable to generate diff analysis: %v", err)
+			}
+
+			diffs, err := analyzer.CompareAllPaths(analysisOne, analysisTwo)
+			if err != nil {
+				logger.Fatalf("unable to generate diff analysis: %v", err)
+			}
+
+			if err = printDiffedPathTable(diffs); err != nil {
+				logger.Fatalf("unable to print diff analysis: %v", err)
+			}
+
+		} else if args[0] == "intersect" {
+			analysisOne, analysisTwo, err := analyzer.HighlightAnalysis(graphs[0], graphs[1], 1)
+			if err != nil {
+				logger.Fatalf("Unable to generate intersect analysis: %v", err)
+			}
+			if err = printPathTable("Common Paths", analysisOne, analysisTwo); err != nil {
+				logger.Fatalf("unable to print intersect analysis: %v", err)
+			}
+		} else if args[0] == "union" {
+			analysisOne, analysisTwo, err := analyzer.HighlightAnalysis(graphs[0], graphs[1], 2)
+			if err != nil {
+				logger.Fatalf("unable to generate union analysis: %v", err)
+			}
+			if err = printPathTable("All Paths", analysisOne, analysisTwo); err != nil {
+				logger.Fatalf("unable to print union analysis: %v", err)
+			}
+		}
+	},
+}
+
+func printDiffedPathTable(diffs []analyzer.DiffedPath) error {
+
+	table := tablewriter.NewWriter(os.Stdout)
+	table.SetAutoWrapText(false)
+
+	table.SetBorders(tablewriter.Border{Left: true, Bottom: true})
+
+	table.SetNoWhiteSpace(true)
+
+	table.SetColumnSeparator("\t\t")
+	table.SetAutoMergeCells(false)
+
+	table.SetHeader([]string{"Path Differences"})
+
+	for _, diff := range diffs {
+		var row []string
+
+		for i, nodeOne := range diff.PathOne {
+
+			if len(row) != 0 {
+				row = append(row, "--->")
+				table.SetColMinWidth(i+1, 90)
```

Comment:
having constants for the formatting will make it easier to change in the future.